File: chroma-league/src/main/java/net/booone/chromaleague/tasks/GameLoadingTask.java
package net.booone.chromaleague.tasks;

import net.booone.chromaleague.hud.animations.LoadingAnimation;
import net.booone.chromaleague.state.RunningState;
import net.booone.razersdk.ChromaRestSDK;

import java.util.logging.Level;
import java.util.logging.Logger;

public record GameLoadingTask(ChromaRestSDK chromaRestSDK,
                              LoadingAnimation loadingAnimation) implements Runnable {
    private static final Logger LOGGER = Logger.getLogger(GameLoadingTask.class.getName());

    @Override
    public void run() {
        if (!RunningState.getRunningGame().getValue()) {
            try {
                chromaRestSDK.createKeyboardEffect(loadingAnimation);
            } catch (Exception ex) {
                LOGGER.log(Level.SEVERE, ex, () -> "Error while refreshing loading animation");
            }
        }
    }
}

File: src/tasks/GameLoadingTask.ts
import {LoadingAnimation} from '../hud/animations/LoadingAnimation.js';
import type {ChromaNativeSDK} from '../razer-sdk/ChromaNativeSDK.js';
import {RunningState} from '../state/RunningState.js';
import {clearErrorSuppression, logErrorSuppressed} from './TaskErrorUtils.js';

const TASK_KEY = 'GameLoadingTask';

export function createGameLoadingTask(chromaNativeSDK: ChromaNativeSDK, loadingAnimation: LoadingAnimation): () => Promise<void> {
    return async () => {
        if (!RunningState.getRunningGame().getValue()) {
            try {
                await chromaNativeSDK.createKeyboardEffect(loadingAnimation);
                clearErrorSuppression(TASK_KEY);
            } catch (ex) {
                logErrorSuppressed(TASK_KEY, ex, 'Error while refreshing loading animation');
            }
        }
    };
}
